csharp/src/Google.Protobuf/Collections/MapField.cs

```diff
@@ -469,7 +479,24 @@ public void WriteTo(CodedOutputStream output, Codec codec)
         [SecuritySafeCritical]
         public void WriteTo(ref WriteContext ctx, Codec codec)
         {
-            foreach (var entry in list)
+            if (ctx.state.CodedOutputStream is not null && ctx.state.CodedOutputStream.IsSerializationDeterministic())
```

Comment:
This can just be:

```csharp
if (ctx.state.CodedOutputStream?.IsDeterministic == true)
```

Reply:
Refactored to `if (ctx.state.CodedOutputStream?.Deterministic ?? false)`